web/src/portico/signup.ts

```diff
@@ -14,20 +15,43 @@ $(() => {
     // some of the jQuery selectors below will return empty lists.
 
     const $password_field = $<HTMLInputElement>("input#id_password, input#id_new_password1");
+
     if ($password_field.length > 0) {
+        // Flag to track first focus out event
+        let firstFocusOutOccurred = false;
+
         $.validator.addMethod(
             "password_strength",
             (value: string) => password_quality(value, undefined, $password_field),
             () => password_warning($password_field.val()!, $password_field),
         );
+
         // Reset the state of the password strength bar if the page
         // was just reloaded due to a validation failure on the backend.
         password_quality($password_field.val()!, $("#pw_strength .bar"), $password_field);
 
-        $password_field.on("input", function () {
-            // Update the password strength bar even if we aren't validating
-            // the field yet.
-            password_quality($(this).val()!, $("#pw_strength .bar"), $(this));
+        // Listen for input event, but only validate after first blur
+        $password_field.on(
+            "input",
+            _.debounce(function (this: HTMLInputElement) {
+                // Update the password strength bar as user types
+                password_quality($(this).val()!, $("#pw_strength .bar"), $(this));
+
+                // Check if first focus out has occurred
+                if (firstFocusOutOccurred) {
+                    // Validate the password field after first blur
+                    $password_field.valid();
+                }
+            }, 500),
+        );
+
+        // Listen for the first focus out (blur) event
+        $password_field.one("blur", () => {
+            // Validate the password field on the first blur
+            $password_field.valid();
+
+            // Set the flag to true after the first blur
```

Comment:
These comments are largely useless; don't say what the code does, that can be read by anyone. Comments are for explaining the logic, not repeating what the code does.

Check out our  [commit guidelines](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html) for more details.